Backpack/src/main/java/net/skyscanner/backpack/contentbubble/BpkContentBubble.kt

```diff
@@ -0,0 +1,224 @@
+    }
+
+    cornerRadiusDrawable.apply {
+      setBounds(0, 0, intrinsicWidth, intrinsicHeight)
+    }
+
+    setUpMasks()
+  }
+
+  override fun draw(canvas: Canvas) {
+    val count = canvas.saveCount
+
+    val width = width.toFloat()
+    val height = height.toFloat()
+
+    val pointerHalfWidth = pointerDrawable.intrinsicWidth / 2
+    val pointerOffset = if (layoutDirection == View.LAYOUT_DIRECTION_RTL) {
+      1 - pointerPosition.offset
+    } else {
+      pointerPosition.offset
+    }
+
+    val yStart = height - pointerDrawable.intrinsicHeight
+    val pointerXStart = width * pointerOffset - pointerHalfWidth
+    val pointerXEnd = width * pointerOffset + pointerHalfWidth
+
+    clipRect.set(0f, yStart, pointerXStart, height)
+    canvas.clipOutRectCompat(clipRect)
+
+    clipRect.set(width, yStart, pointerXEnd, height)
+    canvas.clipOutRectCompat(clipRect)
+
+    super.draw(canvas)
+    canvas.restoreToCount(count)
+
+    paint.xfermode = porterDuffXfermode
+    canvas.drawBitmap(mask, pointerXStart, yStart, paint)
+
+    if (round) {
+      drawRadiusMask(canvas)
+    }
+
+    paint.xfermode = null
+  }
+
+  override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
+    val firstChild = getChildAt(0)
+    if (firstChild == null || !fitContent || firstChild.layoutParams.height >= 0) {
+      super.onMeasure(widthMeasureSpec, heightMeasureSpec)
+    } else {
+      getChildAt(0)?.let {
+        // When fit content is true we add the size of the pointer to the view's height, in order
+        // for the children to take up that size we need to change `WRAP_CONTENT` to `MATCH_PARENT`
+        if (it.layoutParams.height == ViewGroup.LayoutParams.WRAP_CONTENT) {
+          it.layoutParams.height = ViewGroup.LayoutParams.MATCH_PARENT
+        }
+        it.measure(widthMeasureSpec, heightMeasureSpec)
+      }
+
+      super.onMeasure(
+        widthMeasureSpec,
+        MeasureSpec.makeMeasureSpec(
+          getChildAt(0).measuredHeight + pointerDrawable.intrinsicHeight,
+          MeasureSpec.EXACTLY
+        ))
+    }
+  }
+
+  private fun Canvas.clipOutRectCompat(rect: RectF) {
+    if (Build.VERSION.SDK_INT >= 26) {
+      this.clipOutRect(rect)
+    } else {
+      @Suppress("DEPRECATION")
+      this.clipRect(rect, Region.Op.DIFFERENCE)
+    }
+  }
+
+  private fun setUpMasks() {
+    mask = Bitmap.createBitmap(pointerDrawable.intrinsicWidth, pointerDrawable.intrinsicHeight, Bitmap.Config.ARGB_8888)
+    maskCanvas = Canvas(mask)
+
+    radiusMask = Bitmap.createBitmap(cornerRadiusDrawable.intrinsicWidth, cornerRadiusDrawable.intrinsicHeight, Bitmap.Config.ARGB_8888)
+    radiusMaskCanvas = Canvas(radiusMask)
+
+    pointerDrawable.draw(maskCanvas)
+    cornerRadiusDrawable.draw(radiusMaskCanvas)
+  }
+
+  private fun drawRadiusMask(canvas: Canvas) {
+    val width = width.toFloat()
+    val height = height.toFloat()
+    val radiusHeight = cornerRadiusDrawable.intrinsicHeight
```

Comment:
We need to use Bitmap dimensions here rather then drawable. The fact, that it match for now is just a coincide. We draw a bitmap, so we need to use it as a reference.

Reply:
done